### PR TITLE
Simplify partition context

### PR DESCRIPTION
**Pull request by pihme**
## Description

This step eliminates two steps in the partition transition sequence. The steps can be eliminated, because the output of each step is only used in one subsequent step. So basically the steps are merged with the follow-up steps that depend on them.

It also aligns the order of leader and follower steps to make it clearer that follower is a subset of leader.

## Definition of Done

Code changes:
* [X] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/0.25`) to the PR, in case that fails you need to create backports manually.

Testing:
* [X] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

File: broker/src/main/java/io/camunda/zeebe/broker/system/partitions/impl/steps/LogStreamPartitionStep.java [hunk 1]
```diff
@@ -7,35 +7,49 @@
...
-                    .registerComponent(logStream.getLogName(), logStream);
-                openFuture.complete(null);
-              } else {
-                openFuture.completeExceptionally(err);
-              }
-            }));
+              }));
+    } else {
+      openFuture.completeExceptionally(logStorageOrException.getLeft());
+    }
```
Comment (Zelldon) [inline broker/src/main/java/io/camunda/zeebe/broker/system/partitions/impl/steps/LogStreamPartitionStep.java:52]: Could be changed to `ifRightOrLeft(Consumer, Consumer)`

**Review by ChrisKujawa — APPROVED** [review]
Thanks @pihme I like it :rocket: 

I added some suggestions feel free to ignore them :)

**Comment by pihme** [comment]
bors retry

**Pull request merged**